flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/ShowCreateUtil.java

```diff
@@ -87,6 +89,9 @@ public static String buildShowCreateTableRow(
                     String.format(
                             "SHOW CREATE TABLE is only supported for tables, but %s is a view. Please use SHOW CREATE VIEW instead.",
                             tableIdentifier.asSerializableString()));
+        } else if (table.getTableKind() == CatalogBaseTable.TableKind.MATERIALIZED_TABLE) {
```

Comment:
it will be executed from `ShowCreateTable` which is wrong
Moreover header's javadoc for `ShowCreateTable` says that that class is only for `SHOW CREATE TABLE`

At the same time the PR added logic for `SHOW CREATE MATERIALIZED TABLE` it should be completely another class